src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -1157,14 +1166,15 @@ export class SearchSource {
   toExpressionAst({ asDatatable = true }: ExpressionAstOptions = {}): ExpressionAstExpression {
     const searchRequest = this.mergeProps();
     const { body, index, query } = searchRequest;
+    const dataView = typeof index !== 'string' ? index : undefined;
```

Comment:
This check on `index` as to whether its a string or a data view happens enough that it might be abstracted into a function. I'll leave it up to you

Reply:
Good point, I'll add this